packages/ses/docs/lockdown.md

```diff
@@ -366,12 +367,54 @@ the container to exit explicitly, and we highly recommend setting
 - `'none'`: do not install traps for uncaught exceptions. Errors are likely to
   appear as `{}` when they are reported by the default trap.
 
+## `evalTaming` Options
+
+This option only affects the initial compartment!
+
+**Background**: For every compartment including the implied compartment in the
+initial JavaScript realm, there are evaluators `eval` and `Function`.
+The default lockdown behavior isolates all of these evaluators.
+
+Replacing the realm's initial evaluators is not necessary to ensure the
+isolation of guest code because guest code must not run in the initial realm.
+However, replacing these evaluators is the safest and therefore default
+evaluator taming option (`"safeEval"`) because it may limit the harm that guest code can cause if it escapes its assigned compartment.
+
+However, only the exact `eval` function from the initial realm can be used to
+perform direct-eval, which runs in the lexical scope in which the direct-eval syntax appears (direct-eval is a special form rather than a function call).
+The SES shim itself uses direct-eval internally to construct an isolated
+evaluator, so replacing the initial `eval` prevents any subsequent program
+from using the same mechanism to isolate a guest program.
+
+The `"unsafeEval"` option for `evalTaming` leaves the original `eval` in place
+for other isolation mechanisms like isolation code generators that work in
+tandem with SES.
+This option may be useful for web pages with a Content Security Policy
+that excludes `unsafe-eval` or browser extension environments with similar
+restrictions, or development-mode bundling systems that use `eval`.
+In these cases, SES cannot be responsible for maintaining the isolation of
+guest code. If you're going to use `eval`, [Trusted
+Types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/trusted-types) may help maintain security.
+
+The `"noEval"` option emulates a Content Security Policy that excludes
+`unsafe-eval` by replacing all evaluators with functions that throw an
+exception.
```

Comment:
Shouldn't `compartment.evaluate()` also throw in this case?

Reply:
I think so. But if so, then the first sentence above
> This option only affects the initial compartment!

needs revision.

Reply:
Actually thinking about this more, since `compartment.evaluate()` evaluates in the context of the child compartment, `noEval` as implemented currently still only applies to the start compartment, so it is consistent.

This section of the doc should be updated to highlight that only evaluators for the start compartment are neutered, but that child compartments remain with safe evaluators. Those can be tamed explicitly by the program if needed (by just copying the throwing noEval `eval` and `Function` from the start compartment into the global object of the child compartment)

Reply:
> Those can be tamed explicitly by the program if needed (by just copying the throwing noEval eval and Function from the start compartment into the global object of the child compartment)

Or also adding it to the Compartment constructor? Therefore devs can get the noEval in the Compartment without make the initial compartment also using noEval

```js
const c = new Compartment({}, { evalTaming: 'noEval' })
c.evaluate('1+1') // ok
c.globalThis.noEval('1+1') // no
```

Reply:
I can’t think of a reason that would motivate this level of specificity.